### PR TITLE
*: do not use bytespool to allocate coprocessor response.

### PR DESCRIPTION
**Pull request by coocood**
The cost to reuse coprocessor bytes is high, it requires change many level of function, the code become hard to maintain.

And after try to implement reuse, the improvement is not as good as expected.

This PR change the `kv.Response` return value type to `[]byte`, simplify code.

### CODE REVIEW

File: distsql/distsql.go [hunk 1]
```diff
@@ -101,23 +98,21 @@ func (r *selectResult) fetch(ctx goctx.Context) {
...
 		}
 		pr := &partialResult{
 			index:      r.index,
 			fields:     r.fields,
-			reader:     reader,
 			aggregate:  r.aggregate,
 			ignoreData: r.ignoreData,
-			done:       make(chan error, 1),
 		}
-		go pr.fetch()
```
Comment (tiancaiamao) [inline distsql/distsql.go]: Any performance penalty after remove the parallel unmarshal process?

File: kv/kv.go [hunk 2]
```diff
@@ -155,7 +153,7 @@ type Response interface {
 	// Next returns a resultSubset from a single storage unit.
 	// When full result set is returned, nil is returned.
 	// TODO: Find a better interface for resultSubset that can avoid allocation and reuse bytes.
-	Next() (resultSubset io.ReadCloser, err error)
```
Comment (shenli) [inline kv/kv.go]: Is there any benefit to change resultSubset from io.ReadCloser to []byte?

File: distsql/distsql.go (hunk 1 above)
Comment (coocood) [inline distsql/distsql.go]: No, the old code assumes the operation is time-consuming, but it's not actually.

File: kv/kv.go (hunk 2 above)
Comment (coocood) [inline kv/kv.go]: Simplify code.

**Comment by ngaut** [comment]
Nice clean up.

**Comment by ngaut** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**